File: cpp/0345.cpp
class Solution {
private:
    bool isvowel(char c) {
        c = tolower(c);
        return ((c == 'a') 
            || (c == 'e')
            || (c == 'i')
            || (c == 'o')
            || (c == 'u'));
    }
public:
    string reverseVowels(string s) {
        int left = 0;
        int right = s.length() - 1;

        while (true) {
            while (left < s.length() && !isvowel(s[left])) {
                left++;
            }
            while (right > -1 && !isvowel(s[right])) {
                right--;
            }

            if (left < right) {
                swap(s[left], s[right]);
                left++;
                right--;
            } else {
                break;
            }
        }

        return s;
    }
};

File: python/0345.py
class Solution {
public:
    bool isVowels(char c) {
        c = tolower(c);
        return c == 'a' || c == 'e'
            || c == 'i' || c == 'o'
            || c == 'u';
    }
    string reverseVowels(string s) {
        vector<char> c;
        for(int i = 0; i < s.length(); i++)
            if(isVowels(s[i]))
                c.push_back(s[i]);      
        for(int i = 0, j = 0; i < s.length(); i++)
        if(isVowels(s[i]))
        {
            s[i] = c;
            j++;
        }
        return s;
    }
};